src/app/content/practiceQuestions/components/Question.tsx

```diff
@@ -40,25 +36,51 @@ const getChoiceLetter = (value: number) => {
 };
 
 // tslint:disable-next-line: variable-name
-const Question = (props: QuestionProps) => {
+const Question = () => {
   const [selectedAnswer, setSelectedAnswer] = React.useState<PracticeAnswer | null>(null);
+  const [showCorrect, setShowCorrect] = React.useState(false);
+  const question = useSelector(pqSelectors.question);
+  const section = useSelector(pqSelectors.selectedSection);
+  const isSubmitted = useSelector(pqSelectors.isCurrentQuestionSubmitted);
+  const dispatch = useDispatch();
+
+  React.useEffect(() => {
+    return () => {
+      setSelectedAnswer(null);
```

Comment:
i think selectedAnswer is less bad because it will naturally not be found in the new question, we could add a helper
```
const selectedAnswer = question.answers.find(mach(selectedAnswerState));
```

and not bother even clearing it when question changes the helper will just start returning null.

similarly what if instead of a boolean we put the question `| null` as the value of showCorrect, and a helper

```
const showCorrect = showCorrectState === question;
```

that way the only thing that actually changes is the question and we dont have any race conditions